src/solver.jl

```diff
@@ -9,61 +10,51 @@
 
 
 # Policy struct created according to one used in DiscreteValueIteration
-mutable struct FiniteHorizonValuePolicy{Q<:AbstractArray, U<:AbstractMatrix, P<:AbstractMatrix, A, M<:MDP} <: Policy
+mutable struct FiniteHorizonValuePolicy{Q<:AbstractArray, U<:AbstractMatrix, P<:AbstractMatrix, A, W<:FHWrapper} <: Policy
     qmat::Q
     util::U 
     policy::P 
     action_map::Vector{A}
     include_Q::Bool 
-    mdp::M
+    w::W
```

Comment:
I think this should be able to hold any MDP, right? As in #15 a user should be able to define their own finite horizon MDPs and they shouldn't have to use an FHWrapper. Same below.

Reply:
Of course, the solver should throw an error if `HorizonLength(m) = InfiniteHorizon()`.